torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1485,6 +1519,17 @@ def _prep_grads_for_backward(self) -> None:
                 p.grad.size() != p._orig_size  # type: ignore[attr-defined]
                 or p.grad.device != p.device
             ):
+                if p.grad.device != p.device:
+                    assert self.cpu_offload.offload_params, \
+                        "`p.grad.device` and `p.device` should be the same " \
+                        "if not offloading parameters to CPU"
+                can_accumulate_grad = p.grad.device == p.device and \
```

Comment:
this means that currently grad will not be accumulated when cpu offloading is enabled? if so, let's add a comment